Automattic/wp-calypso
# Components: Move SiteIcon to blocks and refactor as function component

This pull request seeks to refactor the SiteIcon component as a function component and move it to the blocks directly. The component has received a number of changes as part of the Site Icon project, and this pull request seeks to finish remaining clean-up tasks without introducing new functionality.

__Testing instructions:__

Verify that there are no regressions in the display of site icons, including icons of various sizes (e.g. compare My Sites sidebar vs. Site Settings preview), and both in cases of an icon existing and not existing for a site.

Also ensure that there are no lingering references to SiteIcon in its previous `components` directory.
